code/go/0chain.net/smartcontract/minersc/handler.go

```diff
@@ -79,19 +81,101 @@ func (msc *MinerSmartContract) GetNodepoolHandler(ctx context.Context, params ur
 }
 
 func (msc *MinerSmartContract) GetMinerListHandler(ctx context.Context, params url.Values, balances cstate.StateContextI) (interface{}, error) {
-	allMinersList, err := msc.GetMinersList(balances)
+
+	var (
+		activeString  = params.Get("active")
+		hasQuery      bool
+		allMinersList = new(MinerNodes)
+		minerQuery    event.Miner
+		miners        []event.Miner
+		err           error
+	)
+
+	if balances.GetEventDB() != nil {
```

Comment:
Smartcontracts should not be using GetEventDb. They should not be able to call the method. This whole block can't exist. What is wrong with keeping the existing code?

Here is the problem, smartcontract are deterministic. They can not depend on external factors, like whether like whether some external database is offline or hacked. 

This should be fixed in [issue 657](https://github.com/0chain/0chain/issues/657)

Reply:
reverted to older handler